src/Database/Type/DecimalType.php

```diff
@@ -45,50 +45,55 @@ class DecimalType extends BaseType implements BatchCastingInterface
     /**
      * Convert integer data into the database format.
      *
-     * @param string|int|float $value The value to convert.
+     * @param mixed $value The value to convert.
      * @param \Cake\Database\Driver $driver The driver instance to convert with.
-     * @return string|null
+     * @return mixed
      * @throws \InvalidArgumentException
      */
-    public function toDatabase($value, Driver $driver): ?string
+    public function toDatabase($value, Driver $driver)
     {
         if ($value === null || $value === '') {
             return null;
         }
-        if (!is_scalar($value)) {
-            throw new InvalidArgumentException(sprintf(
-                'Cannot convert value of type `%s` to a decimal',
-                getTypeName($value)
-            ));
-        }
-        if (is_string($value) && is_numeric($value)) {
+
+        if (is_numeric($value)) {
             return $value;
```

Comment:
Should this string cast float/ints into string?

Reply:
We already specify the binding type as `PDO::PARAM_STR` through the `toStatement()`. So there's no need for explicit casting, let PDO handle the value conversion as needed.